app/src/main/java/com/example/recipeapp/Adapters/ReviewsAdapter.java

```diff
@@ -0,0 +1,69 @@
+package com.example.recipeapp.Adapters;
+
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.databinding.DataBindingUtil;
+import androidx.databinding.ViewDataBinding;
+import androidx.lifecycle.MutableLiveData;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.recipeapp.BR;
+import com.example.recipeapp.R;
+import com.example.recipeapp.viewmodels.ReviewItemViewModel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReviewsAdapter extends RecyclerView.Adapter<ReviewsAdapter.BindableViewHolder> {
+
+    private List<ReviewItemViewModel> data = new ArrayList<>();
+    MutableLiveData<ReviewItemViewModel> itemClicked;
+
+    @NonNull
+    @Override
+    public BindableViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        ViewDataBinding binding = DataBindingUtil.inflate(
+                LayoutInflater.from(parent.getContext()),
+                R.layout.review_item,
+                parent,
+                false
+        );
+        return new BindableViewHolder(binding);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull BindableViewHolder holder, int position) {
+        holder.bind(data.get(position));
+    }
+
+    public void updateItems(List<ReviewItemViewModel> items) {
+        data.addAll(items);
+        notifyDataSetChanged();
+    }
+
+    public void addAtTheBeginning(ReviewItemViewModel item) {
+        data.add(0, item);
```

Comment:
Since this is an array, if we add at the beginning it will O(n), can we use some other dataset such as linkedlist.